app/js/lib/intent-parser/index.js

```diff
@@ -0,0 +1,28 @@
+import ReminderParser from './reminder-parser';
+import QueryParser from './query-parser';
+
+const p = Object.freeze({
+  // Properties
+  parsers: Symbol('parsers'),
+});
+
+export default class IntentParser {
+  constructor() {
+    this[p.parsers] = [
+      new ReminderParser(),
+      new QueryParser(),
+    ];
+
+    window.intentParser = this;
+  }
+
+  parse(phrase = '') {
+    const promises = this[p.parsers]
+      .map((parser) => parser.parse(phrase));
+
+    // Return the first non null parsed result.
+    return Promise.all(promises)
```

Comment:
Would it make sense to join the parse results?  I guess it depends on how fine grained the parsers are - but I'm thinking of this:

"Remind me to build a castle at 5PM tomorrow"
"Am I due to build a castle at 5PM tomorrow?"

These are mostly identical in structure and a common parser could extract 
```
{ 
  action: 'build a castle', 
  time: <timestamp>
}
```
and another could extract `{ type: 'reminder' }` or `{ type: 'affirmative-query' }`  (affirmative indicating a 'yes/no' type response: ("Am I..." "Yes you are due to build a castle tomorrow at 5PM") respectively - if we then join we'd end up with something like:

```
{
  type: 'reminder',
  action: 'build a castle',
  time: <timestamp>
}
&&
{
  type: 'affirmative-query',
  action: 'build a castle',
  time: <timestamp>
}
```

I may be getting ahead of myself with different types of query etc. But I think joining makes sense to reduce duplicate parsing?